### ISSUE TITLE
Hangs with Glide 2.1.1 tests

### ISSUE THREAD

**Issue by Vort** [bochs-emu/Bochs#147]
When `test01.exe` from [glide211.zip](https://github.com/bochs-emu/Bochs/files/13451559/glide211.zip) is executed, simulation hangs.
Hang happens with debug (no optimization) builds of Bochs.
Test files were extracted from [this archive](http://falconfly.3dfx.pl/downloads/webrelease-glide211.zip).

This is the same hang as described in discussion started from [this comment](https://github.com/bochs-emu/Bochs/issues/67#issuecomment-1793376795) in #67.
When `if (step == 0) break;` line is added to `bx_virt_timer_c::timer_handler`, execution goes further, but hangs at another step (however, this time guest hangs, not Bochs itself).

Version: eae6bf939635b690405f006352caf5aebee6ccf9

**Comment by Vort** [comment]
Looks like I have (hack-?)fix for all problems described in this report:
```diff
diff --git a/bochs/iodev/display/voodoo.cc b/bochs/iodev/display/voodoo.cc
index ffc76a158..2312acee1 100644
--- a/bochs/iodev/display/voodoo.cc
+++ b/bochs/iodev/display/voodoo.cc
@@ -1070,7 +1070,8 @@ bool bx_voodoo_1_2_c::update_timing(void)
   BX_INFO(("Voodoo output %dx%d@%uHz", v->fbi.width, v->fbi.height, (unsigned)v->vertfreq));
   v->fbi.swaps_pending = 0;
   v->vtimer_running = 1;
-  bx_virt_timer.activate_timer(s.vertical_timer_id, (Bit32u)s.vdraw.vtotal_usec, 1);
+  if (v->vidclk != 0.0)
+    bx_virt_timer.activate_timer(s.vertical_timer_id, (Bit32u)s.vdraw.vtotal_usec, 1);
   return 1;
 }
```
More tests are needed.

**Comment by stlintel** [comment]
I wondered how vidclk would become 0.0 ...
seems like only from here:

```
    case io_pllCtrl0:
      if (value != v->banshee.io[reg]) {
        v->banshee.io[reg] = value;
        k = (Bit8u)(value & 0x03);
        m = (Bit8u)((value >> 2) & 0x3f);
        n = (Bit8u)((value >> 8) & 0xff);
        v->vidclk = (float)(14318180.0f * ((double)n + 2.0f) / ((double)m + 2.0f) / (double)(1 << k));
        BX_INFO(("Setting VCLK #3 (pllCtrl0) = %.3f MHz", v->vidclk / 1000000.0f));
        if (theVoodooVga != NULL) {
          theVoodooVga->banshee_set_vclk3((Bit32u)v->vidclk);
        }
      }
      break;
```

Do you see the 
BX_INFO(("Setting VCLK #3 (pllCtrl0) = %.3f MHz", v->vidclk / 1000000.0f));
printed ?

**Comment by Vort** [comment]
Maybe other changes are also needed to make proper fix, but at least hang is gone now with cd350d8548484b12fff921e743d621fd262c2837. So I'm closing this report.
